### PR TITLE
Cover: Add aria-label to fixed and repeated image backgrounds

### LINKED ISSUE #49902: Cover Block: fixed background image fails a11y test
### Description

This is related to https://github.com/WordPress/gutenberg/issues/30505 which was fixed and closed, but a slightly different issue.

When a regular background image is used in a cover block, the alt text is correctly added and displayed in an img tag. However, if the "Fixed background" or "Repeated background" options are checked, the img tag switches to a div with role="img". SiteImprove then flags it as not having alt text, because an element with role="img" is supposed to have alt text. This happens regardless of whether or not the image has alt text in the media library.

### Step-by-step reproduction instructions

1. Add a cover image block and give it an image background
2. Under "Media Settings", give it alt text
3. Flip the "Fixed Background" switch on
4. Save and view, then run SiteImprove to test for accessibility

### Screenshots, screen recording, code snippet

Screenshot 2023-04-18 at 10 37 15 AM
Screenshot 2023-04-18 at 10 38 25 AM

### Environment info

WP 6.2 using a custom theme. Gutenberg not installed, just using block editor.
Mac Chrome with SiteImprove extension. 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by carolinan**
## What?

When a cover block has a fixed or repeated background, it uses a div with the attribute `role="img"`.
The goal of this PR is to solve two issues:

1. The `role="img"` should not be used if the image is presentational (decorative) and has no `alt` text.
2. If the image is not presentational but content, then it needs to have both `role="img"` and an alternative text. Because `alt=""` is not a valid option for a `div` with `role="img"`, `aria-label` is used instead.

Closes https://github.com/WordPress/gutenberg/issues/49902

## Why?

To improve accessibility when the fixed background or repeated backgrounds are used.

## How?

In block.json, the alt block attribute used the `img` as the selector and the `alt` as the attribute, so I had to remove these
from the block.json. I was not able to make this work with the `<image`, the `<div>`, the `alt`, and the `aria-label `without removing it.

Previous:
```
		"alt": {
			"type": "string",
			"source": "attribute",
			"selector": "img",
			"attribute": "alt",
			"default": ""
		},
```
In the PR:
```
		"alt": {
			"type": "string",
			"default": ""
		},
```

## Testing Instructions

1. Add a cover block with an image background.
2. Enable the options: Fixed background and Repeated background.
3.  Confirm that the alt text field is still present in the block settings sidebar. Previously, this field would have been hidden.
4. With the alt text field **empty**: Please view the source and confirm that the div that has background image in the style attribute does not have a `role="img"` and does not have an `aria-label`.  Do the same on the front.
5. Add the alt text and view the source. Confirm that the div has the alt text content inside an aria-label, and the role="image" is present. Do the same on the front.
6. Deactivate Fixed background and Repeated background,  and confirm that the image now uses the `img` tag and the `alt` attribute. Do the same on the front.

### Testing Instructions for Screen readers

1. Add a cover block with an image background.
2. Navigate to the block settings sidebar.
3. In the settings tab,  navigate past the Layout panel to the Media settings panel.
4. Enable the two first checkboxes: Fixed background and Repeat background.
5. Navigate past to two steppers: left and top.
6. In the text area, add an alternative text.
7. To confirm that the alt text is used with the image background, read the content of the post: The alt will not be announced while the block is selected for editing.
8. Save and view the post or page on the front.
9. On the front, the cover block itself is not focusable. Confirm that the alt text is announced while reading the full page content.

### CODE REVIEW

**Comment by carolinan** [comment]
Deprecation:
- [x] Supports are as they were before the PR
- [x] Attributes are as they were before the PR
- [x] Save function is as it was before the PR 

Fixtures:
One fixture updated
One fixture added, for cover blocks with fixed background and alternative text.

File: packages/block-library/src/cover/deprecated.js [hunk 1]
```diff
@@ -182,6 +182,277 @@ const v7toV10BlockSupports = {
 	},
 };
 
+// Deprecation for blocks that does not have the aria-label when the image backround is fixed or repeated.
```
Comment (carolinan) [inline packages/block-library/src/cover/deprecated.js]:
```suggestion
// Deprecation for blocks that does not have the aria-label when the image background is fixed or repeated.
```

**Review by aristath — COMMENTED** [review]
I did not do a full review, but from a quick look the code seems reasonale. I'm not 100% sure about the usability of this, as `alt` and `aria-label` arew different.
cc @afercia for some feedback here 🙏 

I feel this is definitely an improvement over what we currently have,  but it would be good to have some additional feedback here

**Comment by carolinan** [comment]
Setting this as draft since I want to check if the deprecation is correct after the massive merge conflict.

File: packages/block-library/src/cover/block.json [hunk 2]
```diff
@@ -19,9 +19,6 @@
 		},
 		"alt": {
```
Comment (draganescu) [inline packages/block-library/src/cover/block.json:20]: I wonder if the name of the attribute is reflective of what it does now :)

**Comment by draganescu** [comment]
Since it corrects invalid markup produced by the block (having an alt attr for a div with role img) I labeled this as a bug fix.

The code looks good 👏🏻 the only question is if this is the way to do it, same as Ari mentioned, is the solution in the PR a recommented way to handle this situation or an ad hoc workaround that works?

**Comment by carolinan** [comment]
>  I labeled this as a bug fix.

I don't disagree but we are only allowed one Type label.

File: packages/block-library/src/cover/block.json (hunk 2 above)
Comment (carolinan) [reply packages/block-library/src/cover/block.json:20]: Maybe the name could be "accessibleText". I'm not sure that is clearer.

**Review by draganescu — APPROVED** [review]
I will approve this because honestly this looks like the only way to achieve a better assistive tech experience for this situation. The code looks good.

**Review by aristath — APPROVED** [review]

**Pull request merged**